java/src/jmh/java/com/mlt/OmtDecoderBenchmark.java

```diff
@@ -0,0 +1,190 @@
+package com.mlt;
+
+import com.mlt.converter.ConversionConfig;
+import com.mlt.converter.FeatureTableOptimizations;
+import com.mlt.converter.MltConverter;
+import com.mlt.converter.mvt.ColumnMapping;
+import com.mlt.converter.mvt.MapboxVectorTile;
+import com.mlt.converter.mvt.MvtUtils;
+import com.mlt.decoder.MltDecoder;
+import com.mlt.metadata.tileset.MltTilesetMetadata;
+import org.apache.commons.lang3.tuple.Pair;
+import org.openjdk.jmh.annotations.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Benchmark for the decoding performance of OpenMapTiles schema based tiles into the MVT and MLT
+ * in-memory representations.
+ */
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@BenchmarkMode(Mode.AverageTime)
+@Threads(value = 1)
+@Warmup(iterations = 5)
+@Measurement(iterations = 5)
+@Fork(value = 1)
+public class OmtDecoderBenchmark {
+    private static Map<Integer, ByteArrayInputStream> encodedMvtTiles = new HashMap<>();
+    private static Map<Integer, byte[]> encodedMltTiles = new HashMap<>();
+    private static Map<Integer, MltTilesetMetadata.TileSetMetadata> tileMetadata = new HashMap<>();
+
+    @Setup
+    public void setup() throws IOException {
+        encodeTile(2, 2, 2);
+        encodeTile(4, 8, 10);
+        encodeTile(5, 16, 21);
+        encodeTile(7, 66, 84);
+        encodeTile(9, 265, 341);
+        encodeTile(10, 532, 682);
+        encodeTile(12, 2132, 2734);
+        encodeTile(14, 8298, 10748);
+    }
+
+    @Setup(Level.Invocation)
+    public void resetInputStreams() {
+        for(var is : encodedMvtTiles.values()) {
+            is.reset();
+        }
+    }
+
+    private void encodeTile(int z, int x, int y) throws IOException {
+        var encodedMvtTile = getMvtFile(z, x, y);
+        encodedMvtTiles.put(z, encodedMvtTile.getLeft());
+
+        var columnMapping = new ColumnMapping("name", ":", true);
+        var columnMappings = Optional.of(List.of(columnMapping));
+        var metadata = MltConverter.createTilesetMetadata(encodedMvtTile.getRight(), columnMappings, true);
+        tileMetadata.put(z, metadata);
+
+        var allowIdRegeneration = true;
+        var allowSorting = false;
+        var optimization = new FeatureTableOptimizations(allowSorting, allowIdRegeneration, columnMappings);
+        var optimizations = Map.of("place", optimization, "water_name", optimization, "transportation",
+                optimization, "transportation_name", optimization, "park", optimization, "mountain_peak",
+                optimization, "poi", optimization, "waterway", optimization, "aerodrome_label", optimization);
```

Comment:
I'm curious: what happens if any of these field names are misspelled or don't exist in the tileset? Will we get an error? I would assume throwing would be ideal to warn the developer that the optimization is not going to work as expected.